cfme/tests/infrastructure/test_host.py

```diff
@@ -312,3 +312,41 @@ def _finalize():
             actual_state, power_state
         ),
     )
+
+
+@test_requirements.general_ui
+def test_infrastructure_hosts_refresh_multi(
+    appliance, setup_provider, provider
+):
+    """
+    Polarion:
+        assignee: prichard
+        casecomponent: Infra
+        caseimportance: low
+        initialEstimate: 1/3h
+        testSteps:
+            1. Navigate to the Compute > Infrastructure > Hosts view.
+            2. Select at least 2 hosts (more if possible) by checking the box in upper left of
+               quadicons.
+            3. Click "Refresh Relationships and Power States" under the Configuration
+               dropdowm, and then click "OK" when prompted.
+        expectedResults:
+            1. Hosts view is displayed.
+            2.
+            3. "Refresh initiated for X Hosts from the CFME Database" is displayed in green
+               banner where "X" is the number of selected hosts. Properties for each host are
+               refreshed.
+    """
+    num_hosts = len(provider.data.get('hosts', {}))
+    if num_hosts < 2:
+        pytest.skip('not enough hosts to run test')
+    hosts_view = navigate_to(provider.collections.hosts, "All")
+    assert hosts_view.is_displayed
+    for h in hosts_view.entities.get_all():
```

Comment:
I'm pretty sure we don't have that many hosts in any provider but what do you think about using `surf_pages=True`?

Reply:
There is a method in the paginationpane widget for select all, I think we should use it here instead of iterating over like this.

Could be considered a separate test of pagination pane, could be parametrized, but I'm not sure that's worth it.

If we don't necessarily care how these items are selected, I'd use pagination select all, because its faster and a one-line call.

Reply:
This test is for refreshing multiple hosts, not all hosts. I'd like to leave the selection of the hosts similliar to the way it is, not using "all". I am going to take a closer look at how I determine the hosts (although similair code exists elsewhere). I could update the test to just use 2 hosts, but I figured we could test more if we had them and it would be a better test. Could add a refresh all test and use the suggested widget. I need to do some investigating as we are beginnig to go in circles with the PR.